composer/datasets/evaluator.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import List
+
+import yahp as hp
+from torchmetrics.classification.accuracy import Accuracy
+from torchmetrics.classification.f_beta import F1
+from torchmetrics.classification.matthews_corrcoef import MatthewsCorrcoef
+from torchmetrics.collections import MetricCollection
+from torchmetrics.metric import Metric
+from torchmetrics.regression.pearson import PearsonCorrcoef
+from torchmetrics.regression.spearman import SpearmanCorrcoef
+
+from composer.core.types import Metrics
+from composer.datasets.dataset_registry import get_dataset_registry
+from composer.datasets.hparams import DataloaderSpec, DatasetHparams
+from composer.models.loss import CrossEntropyLoss
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluatorHparams(hp.Hparams):
+    hparams_registry = {  # type: ignore
+        "eval_dataset": get_dataset_registry(),
+    }
+
+    metric_registry = {
+        "Accuracy": Accuracy,
+        "MatthewsCorrcoef": MatthewsCorrcoef,
```

Comment:
Can we remove "coef" from these and just do something like "MatthewsCorr"? It's less verbose

Reply:
Instead of having to enumerate all the metrics, would it work to have a `metric_name` hparam field, and then do a `metric = getattr(torchmetrics, self.metric_name)`?